lib/matplotlib/axis.py

```diff
@@ -674,7 +678,12 @@ def __init__(self, axes, *, pickradius=15):
         self._major_tick_kw = dict()
         self._minor_tick_kw = dict()
 
-        self.clear()
+        if clear:
+            self.clear()
```

Comment:
I guess I thought `clear` did a _lot_ more than this.  

I think that the doctoring should maybe clarify what the heck "clear on creation" means, or why one would or would not want to do it.  I'm pretty surprised that everything "works" with just setting the convert and units to "None".  If that is all robust (and tests seem to pass!) that is really a great simplification.

Reply:
The thing is that `Axis.clear` is called from `Axes.clear`, sp in that case we don't not have to call it on creation.

Those two attributes are defined by clear, but since they are checked before clear is called now, they must be defined in `__init__`.

I do not really think that calling clear as part of the creation is a very nice design pattern, but I guess it requires quite a bit of an effort to get rid of it.